Removed the potentiometer from the climber subsystem

rakshaaJey opened the pull request:
fix #9

rakshaaJey committed:
Removed the climber potentiometer

rakshaaJey committed:
Removed the potentiometer import

matthew-reynolds commented on src/main/java/frc/robot/subsystems/Climber.java:
```diff
@@ -76,11 +73,4 @@ public void setPiston(Value value) {
-   */
-  public double getDistance() {
-    return (m_pot.getAverageVoltage() / Constants.AIO_MAX_VOLTAGE) * Constants.CLIMBER_POT_SCALE_VALUE;
-  }
```
Instead of deleting this, can you replace it with the CANSparkMax's built-in encoder's value? You will also need to reset the encoder position at the start of the match. Jacob and/or James can help you figure out what the reduction constants should be.

matthew-reynolds reviewed (changes requested):
Can you please also remove the constants from `RobotMap` and `Constants`?

rakshaaJey committed:
deleted constants and added encoders

rakshaaJey committed:
Added a constant for the encoder

rakshaaJey commented:
I added the CANSparkMax's built in encoders and a reset command here. 
ffc4e58

I added the constant here but I don't know the  reduction value so I just put a dummy number
c6c06c3

matthew-reynolds commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -107,6 +107,12 @@
 
+    //#region Climber Details
+
+    public static final double CLIMEBR_ENCODER_INCHES_PER_TICK = 1;
```
Typo, `CLIMEBR` -> `CLIMBER`

Please also leave a `TODO` comment here so we can update the value later

matthew-reynolds commented on src/main/java/frc/robot/subsystems/Climber.java:
```diff
@@ -20,21 +19,26 @@
-    m_pot = new AnalogInput(RobotMap.AIO.CLIMBER_POT);
+
+    m_leftEncoder = m_leftClimberMotor.getEncoder();
+    m_rightEncoder = m_rightClimberMotor.getEncoder();
```
Nitpick: Instead of creating these encoder objects, let's just use `getEncoder()` everywhere.

rakshaaJey committed:
Fixed typos and encoder things

rakshaaJey commented:
Implemented those changes here
7acc1ea

Jacob4649 committed:
Replace non-existent methods in climber with real ones

matthew-reynolds commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -101,6 +101,10 @@
 
+  //#endregion
+
+  //#region Turret Details
```
We now have two "Turret Details" regions. That doesn't seem right.

Jacob4649 replied:
resolved in 3d8d1f5

matthew-reynolds commented on src/main/java/frc/robot/subsystems/Climber.java:
```diff
@@ -19,21 +18,24 @@
-  private final AnalogInput m_pot;
-  private final DoubleSolenoid m_climberSolenoid;
+  private CANSparkMax m_leftClimberMotor, m_rightClimberMotor;
+  private DoubleSolenoid m_climberSolenoid;
```
Please re-add "final"

Jacob4649 replied:
resolved in 3d8d1f5

matthew-reynolds commented on src/main/java/frc/robot/subsystems/Climber.java:
```diff
@@ -19,21 +18,24 @@
-  private final DoubleSolenoid m_climberSolenoid;
+  private CANSparkMax m_leftClimberMotor, m_rightClimberMotor;
+  private DoubleSolenoid m_climberSolenoid;
+
```
Please remove double newline here

Jacob4649 replied:
resolved in 3d8d1f5

matthew-reynolds commented on src/main/java/frc/robot/subsystems/Climber.java:
```diff
@@ -19,21 +18,24 @@
     m_rightClimberMotor = new CANSparkMax(RobotMap.CAN.CLIMBER_RIGHT_SPARKMAX, MotorType.kBrushless);
     m_climberSolenoid = new DoubleSolenoid(RobotMap.PCM.CLIMBER_A, RobotMap.PCM.CLIMBER_B);
-    m_pot = new AnalogInput(RobotMap.AIO.CLIMBER_POT);
+
```
Remove double newline

Jacob4649 replied:
resolved in 3d8d1f5

matthew-reynolds commented on src/main/java/frc/robot/subsystems/Climber.java:
```diff
@@ -19,21 +18,24 @@
     m_leftClimberMotor.setInverted(false);
     m_rightClimberMotor.burnFlash();
     m_leftClimberMotor.burnFlash();
+
```
Remove double newline

Jacob4649 replied:
resolved in 3d8d1f5

matthew-reynolds commented on src/main/java/frc/robot/subsystems/Climber.java:
```diff
@@ -80,9 +82,31 @@ public void togglePiston() {
+    m_rightClimberMotor.getEncoder().setPosition(0.0);
   }
-}
+}
```
Add newline at EOF

Jacob4649 replied:
resolved in 3d8d1f5

matthew-reynolds reviewed (changes requested):
Small changes, basically run the formatter please

Jacob4649 committed:
Fix formatting
